tests/_benderjs/ckeditor/static/tools.js

```diff
@@ -1258,11 +1223,11 @@
 					return function() {
 						var promise = test.apply( this );
 
-						if ( Q.isPromise( promise ) ) {
+						if ( promise ) {
 							promise.then( function() {
 									resume();
 								} )
-								.fail( function( err ) {
+								[ 'catch' ]( function( err ) {
```

Comment:
Why `[ 'catch ]` and not `.catch`? Does it break anything?

Reply:
https://github.com/lahmatiy/es6-promise-polyfill#usage-in-ie9

`catch` is reserved in IEs, I was thinking about adding an alias, so no `[ '' ]` characters are needed, apparently I didn't ask this in comment before.

Reply:
Hmmm, does it mean that every code we create will need to use `[ 'catch' ]` to support IE8? 😭 I think alias could be useful here, instead we are sentenced to using `[ 'catch' ]`. That's bad 😬

Reply:
How often do you need `catch` closure? For simple grep on our whole source code, I don't see it's used extensively - at the end, Javascript is not Python :wink:

Creating an alias for `catch` method would break the whole concept of making this implementation an alias for native `Promise` type. It would significantly complicate this implementation because if we start playing with changing native type, we should return wrapper inheriting/mapping this type.

Also, we will lose something I like to call ubiquitous API (referencing to BDD) where you just need to know the name of the type, and you don't need to read docs - you already know the API because it's the standard one.

Reply:
I was rather talking about `try/catch` statement. `Promise.prototype.catch` is mostly syntactic sugar over this statement with `reject` resolver support. It mostly exists in our test code base.

> OTOH, I would add some information to CKEDITOR.tools.promise docs, that if targeting browsers < IE9, [ 'catch' ] instead of .catch should be used. WDYT?

Yep, it's actually not obvious and may save someone googling time.